src/e2e/java/teammates/e2e/cases/lnp/BaseLNPTestCase.java

```diff
@@ -76,7 +76,10 @@
      *                          (in seconds) for the test endpoint.
      */
     protected void setupSpecification(double errorRateLimit, double meanRespTimeLimit) {
```

Comment:
Ideally, the method should be overridden by child class and the child class will call the builder to return a  `LNPSpecification` instance (in any way they want). Therefore, the method should not specify what variables to pass (i.e., we should not have the two arguments in the method).

Reply:
refactored the `setupSpecification` method to abstract and involve no argument. This leads to some changes in child test classes (override & update method call).